refinery/analysis_manager/utils.py

```diff
@@ -0,0 +1,455 @@
+        logger.debug("Current Pair: %s", curr_pair)
+        if curr_pair.node2:
+            temp_pair[input_keys[0]]['node_uuid'] = curr_pair.node1.uuid
+            temp_pair[input_keys[0]]['pair_id'] = count
+            temp_pair[input_keys[1]]['node_uuid'] = curr_pair.node2.uuid
+            temp_pair[input_keys[1]]['pair_id'] = count
+            ret_list.append(temp_pair)
+            logger.debug("Temp Pair: %s", temp_pair)
+            count += 1
+
+    logger.info("Associating analysis with data set %s (%s)",
+                data_set, data_set.uuid)
+
+    # ANALYSIS MODEL
+    # How to create a simple analysis object
+    if not custom_name:
+        temp_name = "{} {}".format(
+            current_workflow.name,
+            get_aware_local_time().strftime("%Y-%m-%d @ %H:%M:%S")
+        )
+    else:
+        temp_name = custom_name
+
+    summary_name = "None provided."
+
+    analysis = Analysis.objects.create(
+        summary=summary_name,
+        name=temp_name,
+        project=user.profile.catch_all_project,
+        data_set=data_set,
+        workflow=current_workflow,
+        time_start=timezone.now()
+    )
+    analysis.set_owner(user)
+
+    logger.debug("ret_list")
+    logger.debug(json.dumps(ret_list, indent=4))
+
+    # ANALYSIS MODEL
+    # Updating Refinery Models for updated workflow input
+    # (galaxy worfkflow input id & node_uuid)
+    count = 0
+    for samp in ret_list:
+        count += 1
+        for k, v in samp.items():
+            temp_input = WorkflowDataInputMap.objects.create(
+                workflow_data_input_name=k,
+                data_uuid=samp[k]["node_uuid"],
+                pair_id=count)
+            analysis.workflow_data_input_maps.add(temp_input)
+            analysis.save()
+    return analysis
+
+
+def create_tool_analysis(validated_analysis_config):
+    """
+    Create an Analysis instance from a validated analysis config with
+    Tool information
+    :param validated_analysis_config: a dict including the necessary
+    information to create an Analysis that has been validated prior by
+    `analysis_manager.utils.validate_analysis_config`
+    :return: an Analysis instance
+    :raises: RuntimeError
+    """
+
+    # Input list for running analysis
+    common_analysis_objects = fetch_objects_required_for_analysis(
+        validated_analysis_config
+    )
+    custom_name = validated_analysis_config["custom_name"]
+    current_workflow = common_analysis_objects["current_workflow"]
+    data_set = common_analysis_objects["data_set"]
+    user = common_analysis_objects["user"]
+
+    try:
+        tool = tool_manager.models.Tool.objects.get(
+            uuid=validated_analysis_config["toolUuid"]
+        )
+    except (tool_manager.models.Tool.DoesNotExist,
+            tool_manager.models.Tool.MultipleObjectsReturned) as e:
+        raise RuntimeError("Couldn't fetch Tool from UUID: {}".format(e))
+
+    analysis = Analysis.objects.create(
+        summary="blah",
```

Comment:
@jkmarx Great question, It's a field on `BaseResource` model, but not entirely sure of its importance or use, other than we were providing `summary` for the Analyses we created before